### PR TITLE
Add RTCPeerConnection constructor error to spec.

### PR DESCRIPTION
**Pull request by VergeA**
Because WebRTC's RTCPeerConnection interface is disallowed in fenced frames, we need to patch the WebRTC spec to throw an exception in the constructor. This patch also introduces the a "fenced navigable ancestor" algorithm that can be used to determine if a given document is nested in a fenced frame tree. This mechanism is used to trigger the constructor error.

***
<a href="https://pr-preview.s3.amazonaws.com/VergeA/fenced-frame/pull/197.html" title="Last updated on Nov 8, 2024, 9:03 PM UTC (b6479f9)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/fenced-frame/197/01cf852...VergeA:b6479f9.html" title="Last updated on Nov 8, 2024, 9:03 PM UTC (b6479f9)">Diff</a>

### CODE REVIEW

**Review by blu25 — CHANGES_REQUESTED** [review]

File: spec.bs [hunk 1]
```diff
@@ -2474,6 +2474,25 @@ nested traversables in [[HTML]]'s <a
 href=https://html.spec.whatwg.org/multipage/document-sequences.html#top-level-traversables>Top-level
 traversables</a> section.
 
+<div algorithm>
```
[the remark this answers is not in the document]
Comment (VergeA) [reply spec.bs:2477]: Ah, great! I figured there had to be an easier way to do this, but wasn't sure what the magical incantation was. I deleted this whole div, and tweaked the WebRTC section below with your recommendation instead.

File: spec.bs [hunk 2]
```diff
@@ -3918,6 +3937,24 @@ at the expense of some utility.
   </wpt>
 </div>
 
+<h3 id=webrtc-monkeypatch>WebRTC</h3>
+
+The [[WEBRTC]] specification defines "ECMAScript APIs in WebIDL to allow media and generic 
+application data to be sent to and received from another browser or device implementing the 
+appropriate set of real-time protocols." The interface which facilitates connections to peers
```
[the remark this answers is not in the document]
Comment (VergeA) [reply spec.bs]: Done, already had rewrap installed (must have been in the Chromium vscode guide).

**Review by blu25 — APPROVED** [review]

**Comment by VergeA** [comment]
Sorry, just realized I should have requested @domfarolino's review first (my brain saw the merge button and went on autopilot). I can revert this if necessary, or we can sync up later to address comments in a follow-up.

File: spec.bs [hunk 3]
```diff
@@ -3918,6 +3918,24 @@ at the expense of some utility.
...
+WebRTC, is disallowed in fenced frames.
+
+<div algorithm=webrtc-constructor>
+  Modify the {{RTCPeerConnection}} {{RTCPeerConnection/constructor}} algorithm to add new first and
+  second steps that read:
+
+  1. Let |navigable| be [=this=]'s [=relevant global object=]'s [=Window/navigable=].
+
+  1. If |navigable|'s [=traversable navigable=] is a [=fenced navigable container/fenced
+     navigable=], throw a {{NotAllowedError}} {{DOMException}}.
```
Comment (domfarolino) [inline spec.bs:3936]: `[=exception/throw=]`

File: spec.bs [hunk 4]
```diff
@@ -3918,6 +3918,24 @@ at the expense of some utility.
...
+application data to be sent to and received from another browser or device implementing the
+appropriate set of real-time protocols." The interface which facilitates connections to peers is
+{{RTCPeerConnection}}. Construction of this interface, and therefore connection to peers via
+WebRTC, is disallowed in fenced frames.
+
+<div algorithm=webrtc-constructor>
+  Modify the {{RTCPeerConnection}} {{RTCPeerConnection/constructor}} algorithm to add new first and
+  second steps that read:
+
+  1. Let |navigable| be [=this=]'s [=relevant global object=]'s [=Window/navigable=].
```
Comment (domfarolino) [inline spec.bs:3933]: `|navigable|` can be null in the detached document case.

Comment (VergeA) [reply spec.bs:3933]: I addressed this in #199.

File: spec.bs (hunk 3 above)
Comment (VergeA) [reply spec.bs:3936]: I addressed this in #199.

**Pull request merged**